pkg/dataobj/internal/util/slicegrow.go

```diff
@@ -0,0 +1,24 @@
+package slicegrow
+
+import "slices"
+
+// GrowToCap grows the slice to at least n elements total capacity.
+// It is an alternative to slices.Grow that increases the capacity of the slice instead of allowing n new appends.
+// This is useful when the slice is expected to have nil values.
+func GrowToCap[Slice ~[]E, E any](s Slice, n int) Slice {
+	return slices.Grow(s, max(0, n-len(s)))
+}
+
+func CopyInto[Slice ~[]E, E any](dst Slice, src Slice) Slice {
```

Comment:
tiny nit: you can probably drop the `Into` parts for CopyInto/CopyStringInto to align with the naming of the `copy` builtin. (Not a big deal though, just saves a few characters)